### ISSUE TITLE
rpc: "id" should be required

### ISSUE THREAD

**Issue by ethers** [ethereum/go-ethereum#924]
https://github.com/ethereum/cpp-ethereum/issues/1841

Go is assuming "id" instead of require it.

**Comment by debris** [comment]
Missing ids **cannot** lead to issues with batch requests. It's unrelated. Although it would be good to to follow jsonrpc standard.

**Comment by obscuren** [comment]
Agreed

**Comment by ethers** [comment]
Thanks debris: I removed comment about batch requests.
Thanks obscuren: for agreeing it would be good for Geth to follow jsonrpc standard.

**Comment by tgerring** [comment]
A fix is applied in 5c65404

Before:

```
$ curl -X POST --data '[{"jsonrpc":"2.0","method":"eth_getBalance","params":["0x0000000000000000000000000000000000000000","latest"],"id":5148},{"jsonrpc":"2.0","method":"eth_getBalance","params":["0x0000000000000000000000000000000000000000","latest"]},{"jsonrpc":"2.0","method":"eth_getBalance","params":["0x0000000000000000000000000000000000000000","latest"],"id":5150}]' http://127.0.0.1:8545
[
	{
		"id": 5148,
		"jsonrpc": "2.0",
		"result": "0x0b7abc627050305ae9ecbb6cb75782f3c93a"
	},
	{
		"id": null,
		"jsonrpc": "2.0",
		"result": "0x0b7abc627050305ae9ecbb6cb75782f3c93a"
	},
	{
		"id": 5150,
		"jsonrpc": "2.0",
		"result": "0x0b7abc627050305ae9ecbb6cb75782f3c93a"
	}
]
```

After:

```
$ curl --X POST --data '[{"jsonrpc":"2.0","method":"eth_getBalance","params":["0x0000000000000000000000000000000000000000","latet"],"id":5148},{"jsonrpc":"2.0","method":"eth_getBalance","params":["0x0000000000000000000000000000000000000000","latest"]},{"jsonrpc":"2.0","method":"eth_getBalance","params":["0x0000000000000000000000000000000000000000","latest"],"id":5150}]' http://127.0.0.1:8545
[
	{
		"id": 5150,
		"jsonrpc": "2.0",
		"result": "0x0b7abc627050305ae9ecbb6cb75782f3c93a"
	},
	{
		"id": 5148,
		"jsonrpc": "2.0",
		"result": "0x0b7abc627050305ae9ecbb6cb75782f3c93a"
	}
]
```
